### PR TITLE
Handle very short packets

### PR DESCRIPTION
**Pull request by rbtcollins**
Yay quickcheck.

### CODE REVIEW

File: python/subunit/tests/test_test_protocol2.py [hunk 1]
```diff
@@ -17,6 +17,10 @@
 from io import BytesIO
 import datetime
 
+from hypothesis import given
+# from hypothesis import Settings, Verbosity
+# Settings.default.verbosity = Verbosity.verbose
```
Comment (jml) [inline python/subunit/tests/test_test_protocol2.py]: No need for comments out code

**Comment by jml** [comment]
Hypothesis creates a .hypothesis directory, which has to be removed by distclean but currently is not, hence failing tests. Please fix this, kill the commented out code, and merge when ready.

File: python/subunit/tests/test_test_protocol2.py (hunk 1 above)
Comment (rbtcollins) [inline python/subunit/tests/test_test_protocol2.py]: So there's no need for it, but its a useful hint for this new thing I think. Do you object to it?

**Comment by rbtcollins** [comment]
@jml ping

File: python/subunit/tests/test_test_protocol2.py (hunk 1 above)
Comment (jml) [inline python/subunit/tests/test_test_protocol2.py]: Mildly. Looking up docs is easy and reading code is relatively hard, so I'm all for killing the clutter. If you decide to keep the comment, I suggest adding `# To get more verbose output:` or similar (assuming that is what this code does!)

**Comment by jml** [comment]
Up to you re comment. Merge when ready.

File: python/subunit/tests/test_test_protocol2.py (hunk 1 above)
Comment (rbtcollins) [inline python/subunit/tests/test_test_protocol2.py]: added a prelude - thans.

**Pull request closed without merging**